plugins/wdc/wdc-nvme.c

```diff
@@ -9112,29 +9093,46 @@ static int wdc_get_fw_act_history_C2(nvme_root_t r, struct nvme_dev *dev,
 		nvme_show_status(ret);
 
 	if (!ret) {
-		/* parse the data */
+		/* Get the log page data and verify the GUID */
 		fw_act_history_log = (struct wdc_fw_act_history_log_format_c2 *)(data);
-		tot_entries = le32_to_cpu(fw_act_history_log->num_entries);
 
-		if (tot_entries > 0) {
-			/* get the FW customer id */
-			if (!wdc_is_sn861(device_id)) {
-				cust_id = wdc_get_fw_cust_id(r, dev);
-				if (cust_id == WDC_INVALID_CUSTOMER_ID) {
-					fprintf(stderr,
-						"%s: ERROR: WDC: invalid customer id\n",
-						__func__);
-					ret = -1;
-					goto freeData;
+		for (i = 0; i < 16; i++) {
```

Comment:
For the loop you could use `WDC_C2_GUID_LENGTH` instead of `16`.  Or replace the whole loop with `memcmp()`:
```
c2GuidMatch = !memcmp(ocp_C2_guid, fw_act_history_log->log_page_guid, WDC_C2_GUID_LENGTH);
```

Reply:
Changed to use memcmp.